ui/ducks/send/send.test.js

```diff
@@ -1,534 +1,1544 @@
-      await store.dispatch(
-        updateSendToken({ address: '0x000', symbol: 'DAI' }),
-      );
-      expect(getPrimaryCurrency(store.getState())).toBe('DAI');
-    });
+        const result = sendReducer(initialState, action);
 
-    it('getSendTokenContract should return contract code from EVM', async () => {
-      global.eth = {
-        contract: () => ({
-          at: jest.fn((address) => {
-            if (address === '0x000') {
-              return '0x000';
-            }
-            return '0xf';
-          }),
-        }),
-      };
-      await store.dispatch(
-        updateSendToken({ address: '0x000', symbol: 'DAI' }),
-      );
-      expect(getSendTokenContract(store.getState())).toBe('0x000');
+        expect(result.gas.mode).toStrictEqual('ADVANCED');
+      });
     });
 
-    it('getSendAmount should return amount from state', async () => {
-      await store.dispatch(updateSendAmount('0x0'));
-      expect(getSendAmount(store.getState())).toBe('0x0');
+    describe('updateRecipientUserInput', () => {
+      it('should update recipient user input with payload', () => {
+        const action = {
+          type: 'send/updateRecipientUserInput',
+          payload: 'user input',
+        };
+
+        const result = sendReducer(initialState, action);
+
+        expect(result.recipient.userInput).toStrictEqual(action.payload);
+      });
     });
 
-    it('getSendHexData should return amount from state', async () => {
-      await store.dispatch(updateSendHexData('0x0'));
-      expect(getSendHexData(store.getState())).toBe('0x0');
+    describe('validateRecipientUserInput', () => {
+      it('should set recipient error and warning to null when user input is', () => {
+        const noUserInputState = {
+          recipient: {
+            mode: 'MY_ACCOUNTS',
+            userInput: '',
+            error: 'someError',
+            warning: 'someWarning',
+          },
+        };
+
+        const action = {
+          type: 'send/validateRecipientUserInput',
+        };
+
+        const result = sendReducer(noUserInputState, action);
+
+        expect(result.recipient.error).toBeNull();
+        expect(result.recipient.warning).toBeNull();
+      });
+
+      it('should error with an invalid address error when user input is not a valid hex string', () => {
+        const tokenAssetTypeState = {
+          ...initialState,
+          recipient: {
+            userInput: '0xValidateError',
+          },
+        };
+        const action = {
+          type: 'send/validateRecipientUserInput',
+          payload: {
+            chainId: '',
+            tokens: [],
+          },
+        };
+
+        const result = sendReducer(tokenAssetTypeState, action);
+
+        expect(result.recipient.error).toStrictEqual('invalidAddressRecipient');
+      });
+
+      // TODO: This seems off, is this suppose to be a networks that dont implement checksum? But the user input is not valid hex in the first place.
```

Comment:
Check

Reply:
Prod on non default network:
![image](https://user-images.githubusercontent.com/4448075/119529589-da6f7f80-bd47-11eb-916c-8d5e5d03329c.png)

![image](https://user-images.githubusercontent.com/4448075/119529663-ee1ae600-bd47-11eb-8461-c9a210ada180.png)

Perhaps the logic is screwed up such that its treating invalid characters (non hex chars) as invalid hex input instead of attempting ENS? I agree though that this logic needs improvement once we get the refactor landed.